rebar.config

```diff
@@ -6,15 +6,17 @@
     {exometer_core, "1.5.7"},
     {exometer_influxdb, "0.6.0"},
     {cut,           "1.0.3"},
-    {dtlsex,        {git, "git://github.com/RoadRunnr/dtlsex.git", {branch, "master"}}},
+    {dtlsex,        {git, "https://github.com/RoadRunnr/dtlsex.git", {branch, "master"}}},
     {regine,        "1.0.0"},
-    {ergw_aaa,      {git, "git://github.com/travelping/ergw_aaa.git", {tag, "3.6.10"}}},
-    {pcapng,        {git, "git://github.com/travelping/pcapng.git", {branch, "master"}}},
+    {ergw_aaa,      {git, "https://github.com/travelping/ergw_aaa.git", {tag, "3.6.15"}}},
+    {pcapng,        {git, "https://github.com/travelping/pcapng.git", {branch, "master"}}},
     {cowboy,        "2.9.0"},
     {pbkdf2,        {git, "https://github.com/emqx/erlang-pbkdf2.git", {tag, "2.0.4"}}},
     {jsx,           "3.1.0"},
     {parse_trans,   "3.3.0"},
-    {hackney,       "1.17.4"}
+    {hackney,       "1.18.1"},
+    {recon,         "2.5.2"},
```

Comment:
If you want add those deps into top level and include it into releases, you also need add them into https://github.com/travelping/capwap/blob/master/src/capwap.app.src

Reply:
why?

Reply:
Well, because those applications will be not defined in releases, which means you can't call them directly. In other words - you will not be able to use them in production. More info here https://www.erlang.org/doc/design_principles/applications.html

Reply:
i can launch `observer_cli:start().` in the remote-shell in that other project - https://github.com/travelping/ergw_internal/blob/master/rebar.config#L52 - and it works without any x.app.src reference for any of those two prod-deps.

so, i do not get your point.

Reply:
Well, you can run in erGW because as you see here is used the same approach what you was remove here, this mean the deps was added into specific profile e.g. in prod profile, then those deps was loaded automatically, but if you try change approach and try add deps by default without add deps into application, when you will try create a release and then run application - you will be not able to run deps, because they are not included automatically into project, you need put into application section. in other words, the approach that you use and that is used in erGW is completely different and, accordingly, you need to configure and prepare differently. You can see what I mean if you will try run now `rebar3 release as prod tar` and then try run project like `_build/default/rel/capwap/bin/capwap-0.0.0 console` - you will see that `observer_cli` is not a part of release and you will not be able to run this application.

Reply:
so … it has nothing to do with x.app.src.

the missing part is an entry in `relx`. that satisfy the release-bundler and includes the files into the final release.

Reply:
> so … it has nothing to do with x.app.src.

Wrong. As you see you already use in `release` the `capwap` - if you will add into `x.app.src` you will not need to touch `relx`... From an architectural point of view, this is not the right approach. Since the applications that are used are not explicitly specified in the application to which they refer. The dependency is not related to the release, but to the application. You need just update `x.app.src` as I write to you few hours ago.

Reply:
@vkatsuba what has the capwap-application to do with observer_cli?

Reply:
@vkatsuba so, observer_cli is for sure not part of capwap application.

so, its part of a release (according to https://www.erlang.org/doc/design_principles/release_structure.html) and with the changes i provided to have the reference of observer_cli and recon in the relx part of the rebar.config, everything is settled. i have checked both local build osx releases as well as docker builds: observer_cli is always included in the release and thus available at runtime at will as soon as diagnosis sessions are required.

Reply:
If you use deps or include any deps into application - this is a part of application, because you expect to use this deps with main application and there is absolutely no difference whether you run it separately or not, you will not use `cli` without `capwap` etc. The https://github.com/erlware/relx is a part of `rebar3`, the `rebar3` is not a part of OTP, this is just a tool for build Erlang/OTP projects. So... again - you can use profile approach, default approach or including into release directly as you try to do it now but each of the approaches is used for specific purposes at the same time if we told about including release applications from OTP and from deps - this is are two different entities and used for different points. You can use any approach what you like, but from architectural perspective this is not ok. In first implementation this was not working at all. Now this is working but used in wrong place - yes you can include application into relx etc. but you will have totally same result if you will update `.src` file as I write yesterday. So, to you `¯\_(ツ)_/¯`.